lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -3255,3 +3255,62 @@ LowererMDArch::LowerEHRegionReturn(IR::Instr * insertBeforeInstr, IR::Opnd * tar
     // return the last instruction inserted
     return retInstr;
 }
+
+IR::Opnd* LowererMDArch::IsOpndNegZero(IR::Opnd* opnd, IR::Instr* instr)
+{
+    IR::Opnd * isNegZero = IR::RegOpnd::New(TyInt32, this->m_func);
+    IR::Opnd *src = opnd;
+
+    if (opnd->IsFloat32())
+    {
+        src = IR::RegOpnd::New(TyFloat64, this->m_func);
+        instr->InsertBefore(IR::Instr::New(LowererMD::MDConvertFloat32ToFloat64Opcode, src, opnd, this->m_func));
+    }
+    Assert(src->IsFloat64());
+
+    // Use UInt64 comparison between the opnd to check and negative zero constant.
+    // For this we have to convert opnd which is a double to uint64.
+    // First create a stack location and store the double value in the stack location.
+    // Read back as uint64 from the stack location.
+
+    // Create an f64 stack location
+    IR::SymOpnd *symOpnd = IR::SymOpnd::New(StackSym::New(TyFloat64, this->m_func), TyFloat64, this->m_func);
+    this->m_func->StackAllocate(symOpnd->m_sym->AsStackSym(), sizeof(double));
+
+    // LEA r3, stackSym
+    IR::RegOpnd *addrOpnd = IR::RegOpnd::New(TyMachReg, this->m_func);
+    IR::Instr *lea = IR::Instr::New(Js::OpCode::LEA, addrOpnd, symOpnd, this->m_func);
+    instr->InsertBefore(lea);
+
+    // MOVSD [stackSym], src
+    IR::IndirOpnd *indirOpnd = IR::IndirOpnd::New(addrOpnd, 0, TyFloat64, this->m_func);
+    IR::Instr *movsd = IR::Instr::New(Js::OpCode::MOVSD, indirOpnd, src, this->m_func);
```

Comment:
Thanks @MikeHolman , updated the diff. x86 still requires the lengthy lowering. I saw that SSE4 has PEXTRD which will help generate better code for x86, but I haven't updated the code for this yet.